sdk/tests/conformance2/rendering/blitframebuffer-outside-readbuffer.html

```diff
@@ -0,0 +1,299 @@
+                    testPassed("pixel at [" + jj + ", " + ii + "] is (" + color + "). It is correct!");
+                } else {
+                    testFailed("pixel at [" + jj + ", " + ii + "] should be (" + expectedColor + "), but the actual color is (" + color + ")");
+                }
+            }
+        }
+
+        // The 2nd round test: blit read framebuffer to the image in draw framebuffer
+        // Only one direction of the read region have pixels outside of the read buffer
+        var tests = [
+             [-1, 0], // clamp to the left edge of the read buffer
+             [0, -1], // clamp to the bottom edge of the read buffer
+             [1, 0],  // clamp to the right edge of the read buffer
+             [0, 1]   // clamp to the top edge of the read buffer
+        ];
+        for (var i = 0; i < 4; ++i) {
+            debug("");
+            switch (i) {
+                case 0: debug("clamp to the left edge of the read buffer"); break;
+                case 1: debug("clamp to the bottom edge of the read buffer"); break;
+                case 2: debug("clamp to the right edge of the read buffer"); break;
+                case 3: debug("clamp to the top edge of the read buffer"); break;
+            }
+            gl.bindFramebuffer(gl.READ_FRAMEBUFFER, fbo_read);
+            gl.bindFramebuffer(gl.DRAW_FRAMEBUFFER, fbo_draw);
+            var srcX = tests[i][0];
+            var srcY = tests[i][1];
+            var offset = dstStart - srcStart;
+            gl.blitFramebuffer(srcX, srcY, srcX + width_read, srcY + height_read,
+                               srcX + offset, srcY + offset, srcX + offset + width_read, srcY + offset + height_read,
+                               gl.COLOR_BUFFER_BIT, gl.LINEAR);
+
+            // Read pixels and check the correctness.
+            var pixels = new Uint8Array(width_draw * height_draw * 4);
+            gl.bindFramebuffer(gl.READ_FRAMEBUFFER, fbo_draw);
+            gl.readPixels(0, 0, width_draw, height_draw, gl.RGBA, gl.UNSIGNED_BYTE, pixels);
+            for (var ii = srcY + offset; ii < srcY + offset + height_read; ++ii) {
+                for (var jj = srcX + offset; jj < srcX + offset + width_read; ++jj) {
+                    var loc = ii * width_draw + jj;
+                    var color = [pixels[loc * 4], pixels[loc * 4 + 1], pixels[loc * 4 + 2], pixels[loc * 4 + 3]];
+                    var expectedColor = ref[loc];
+                    // We may need to covert the color space for pixels in blit region
+                    if ((readbufferHasSRGBImage ^ drawbufferHasSRGBImage)) {
+                        if (drawbufferHasSRGBImage) {
+                            expectedColor = linearToSRGB(expectedColor);
+                        } else {
+                            expectedColor = sRGBToLinear(expectedColor);
+                        }
+                    }
+                    if (checkPixel(color, expectedColor) == true) {
```

Comment:
In a future CL, it might be better to try to use wtu.checkCanvasRect for each individual pixel rather than hand-writing the comparison function, if that would reduce the amount of code.